bundles/org.openhab.binding.modbus.sunspec/src/main/java/org/openhab/binding/modbus/sunspec/internal/dto/InverterModelBlock.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.modbus.sunspec.internal.dto;
+
+import java.util.Optional;
+
+/**
+ * Model for SunSpec compatible inverter data
+ *
+ * @author Nagy Attila Gabor - Initial contribution
+ *
+ */
+public class InverterModelBlock {
+
+    /**
+     * Type of inverter (single phase, split phase, three phase)
+     */
+    public Integer phaseConfiguration;
+
+    /**
+     * Length of the block in 16bit words
+     */
+    public Integer length;
+
+    /**
+     * AC Total Current value
+     */
+    public Integer acCurrentTotal;
+
+    /**
+     * AC Current scale factor
+     */
+    public Short acCurrentSF;
+
+    /**
+     * AC Power value
+     */
+    public Short acPower;
+
+    /**
+     * AC Power scale factor
+     */
+    public Short acPowerSF;
+
+    /**
+     * AC Frequency value
+     */
+    public Integer acFrequency;
+
+    /**
+     * AC Frequency scale factor
+     */
+    public Short acFrequencySF;
+
+    /**
+     * Apparent power
+     */
+    public Optional<Short> acApparentPower;
```

Comment:
Why use an `Optional` here? Since `Optional` is not serializable it shpuld  be avoided as field in DTOs.

Reply:
These classes are not intended to be serialized. They purpose is to server as a middle representation layer between the low level binary types and the high level OH data structures.